src/jpegxl.imageio/jxloutput.cpp

```diff
@@ -365,6 +367,110 @@ JxlOutput::write_tiles(int xbegin, int xend, int ybegin, int yend, int zbegin,
 
 
 
+bool
+JxlOutput::save_metadata(ImageSpec& m_spec, JxlEncoderPtr& encoder)
+{
+    DBG std::cout << "JxlOutput::save_metadata()\n";
+
+    // Write EXIF info
+    std::vector<char> exif = { 0, 0, 0, 0 };
+    encode_exif(m_spec, exif);
+
+    // Write XMP packet, if we have anything
+    std::string xmp = encode_xmp(m_spec, true);
+
+    // Write IPTC IIM metadata tags, if we have anything
+    std::vector<char> iptc;
+    encode_iptc_iim(m_spec, iptc);
+
+    bool use_boxes = m_spec.get_int_attribute("jpegxl:use_boxes", 1) == 1;
+    int compress_boxes
+        = m_spec.get_int_attribute("jpegxl:compress_boxes", 0) == 0
+              ? 0
+              : 1;  // TODO: change to 1 for compression by default
+
+    if (use_boxes) {
+        if (JXL_ENC_SUCCESS != JxlEncoderUseBoxes(m_encoder.get())) {
+            errorfmt("JxlEncoderUseBoxes() failed.\n");
+            return false;
+        }
+        DBG std::cerr << "JxlEncoderUseBoxes() ok\n";
+
+        // Exif
+        std::vector<uint8_t>* exif_data = nullptr;
+        if (!exif.empty())
+            exif_data = reinterpret_cast<std::vector<uint8_t>*>(&exif);
+
+        // XMP
+        std::vector<uint8_t> xmp_data;
+
+        if (!xmp.empty()) {
+            xmp_data.insert(xmp_data.end(), xmp.c_str(),
+                            xmp.c_str() + xmp.length());
+        }
+
+        // IPTC
+        std::vector<uint8_t>* iptc_data = nullptr;
+        if (!iptc.empty()) {
+            static char photoshop[] = "Photoshop 3.0";
+            std::vector<char> head(photoshop,
```

Comment:
Wouldn't be better to use std::vector<uint8_t> ?

Reply:
I can't.
everything from OpenImageIO like encode_iptc_iim(m_spec, iptc); use std::vector<char>

Reply:
I was not precise enough.
I meant to use std::vector<uint8_t> instead of std::vector<char>

Reply:
Understood. Yes, especially for the UTF-8 metadata handling it would be better choice.